includes/class-wc-payments-checkout.php

```diff
@@ -397,32 +395,29 @@ function () use ( $payment_fields ) {
 			}
 
 			// Output the form HTML.
-			?>
-			<?php if ( ! empty( $this->gateway->get_description() ) ) : ?>
+			if ( ! empty( $this->gateway->get_description() ) ) : ?>
 				<p><?php echo wp_kses_post( $this->gateway->get_description() ); ?></p>
-			<?php endif; ?>
+				<?php
+			endif;
 
-			<?php if ( WC_Payments::mode()->is_test() ) : ?>
+			if ( WC_Payments::mode()->is_test() && false !== $this->gateway->get_payment_method()->get_testing_instructions() ) :
```

Comment:
I updated this so that the `div.testmode-info` element will be shown only if test instructions are present - otherwise it's displayed even on Klarna, Affirm, etc.